packages/rpc-provider/src/ws/index.ts

```diff
@@ -174,13 +180,21 @@ export class WsProvider implements ProviderInterface {
     return this.#isReadyPromise;
   }
 
+  public get endpoint (): string {
+    return this.#endpoints[this.#endpointIndex];
+  }
+
   /**
    * @description Returns a clone of the object
    */
   public clone (): WsProvider {
     return new WsProvider(this.#endpoints);
   }
 
+  protected selectEndpointIndex (endpoints: string[]): number {
```

Comment:
Not sure why we are passing through the param here from the usage, i.e. line 206 below - should we not just use `this.#endpoints` here?

Additionally, I'm actually unsure about the utility here of having this function. Since it is only used in the one place (like 206 below), not sure why it is extracted since it doesn't actually set any values itself. So by the definition here -

1. it is external to the class - which means it is callable on the public interface
2. calling it external actually no no impact, i.e. doesn't set anything?

So circling back, I'm either missing something of don't see why this is now on the public interface here since it is a best (if not called below) a no-op.

Reply:
The general idea is to allow subclasses to provide their own logic for endpoint selection (e.g. the one with lowest ping latency, the one with least previous errors per unit of time, ..).
It takes `endpoints` as parameters as it looks to me that having it as a field is an implementation detail that could evolve over time. If it was ok to break compatibility, I probably would have introduced some separate interface to model this.